### PR TITLE
[feat] Add SDK support to React 18

### PR DESCRIPTION
**Pull request by rafael-franca**
## What's the purpose of this pull request?
In React 18 the FC type isn't includes the children prop, and the BaseStore now is based in this version.

## References
https://blog.logrocket.com/upgrading-react-18-typescript/

### CODE REVIEW

**Review by igorbrasileiro — COMMENTED** [review]
Drop the FC in favor of function parameter props.

File: packages/sdk/package.json [hunk 1]
```diff
@@ -39,13 +39,13 @@
     }
   ],
   "peerDependencies": {
-    "react": "^17.0.2"
+    "react": "^17.0.0"
```
[the remark this answers is not in the document]
Comment (rafael-franca) [reply packages/sdk/package.json]: OK

File: packages/sdk/src/search/Provider.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ type Props = SearchState & {
   itemsPerPage: number
 }
 
-export const Provider: FC<Props> = ({
+export const Provider: FC<PropsWithChildren<Props>> = ({
```
Comment (igorbrasileiro) [inline packages/sdk/src/search/Provider.tsx:20]: What do you think about this suggestion?

```tsx
export const Provider = ({ ...props /* current object properties*/}: PropsWithChildren<Props>) => { /* current component body here */}
```

**Review by igorbrasileiro — COMMENTED** [review]
Why you didn't bump those packages to react 18?

**Comment by rafael-franca** [comment]
> Why you didn't bump those packages to react 18?

I don't think need to bump react 18 for only for a type change.

File: packages/sdk/src/search/Provider.tsx (hunk 2 above)
Comment (rafael-franca) [reply packages/sdk/src/search/Provider.tsx]: I like it! Is more clear way. I already changed!

**Review by icazevedo — APPROVED** [review]
LGTM

**Review by igorbrasileiro — COMMENTED** [review]
LGTM, but doesn't need to bump the react to 18, does? Also the @types/react@18 @types/react-dom@18

**Comment by rafael-franca** [comment]
> LGTM, but doesn't need to bump the react to 18, does?

No, because the prop that was changed (from `FC` to `PropsWithChildren`) exists in both versions of React. Change the version will create a unnecessary breaking change.

**Comment by emersonlaurentino** [comment]
@rafael-franca @tlgimenes @icazevedo what's needed to merge this pr?

**Comment by icazevedo** [comment]
> @rafael-franca @tlgimenes @icazevedo what's needed to merge this pr?

Nothing. I think we should rebase and merge it.

**Review by emersonlaurentino — APPROVED** [review]

**Comment by filipewl** [comment]
It looks like the changes here were already applied in the new #1352.

**Comment by rafael-franca** [comment]
> It looks like the changes here were already applied in the new #1352.

Finally! 🎉

**Pull request closed without merging**